source/simulator/core.cc

```diff
@@ -1976,6 +1976,26 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
+        {
+          // Identical to IMPES except does not solve Stokes equation
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
+       
+          std::string fileName = parameters.input_directory + "/" + "solution-" + Utilities::int_to_string(timestep_number, 5) + ".mesh";
+          triangulation.load(fileName.c_str());
+          parallel::distributed::SolutionTransfer<dim, LinearAlgebra::BlockVector> sol_trans(dof_handler);
+          sol_trans.deserialize (solution);
+          //triangulation.clear();
```

Comment:
> Where is the right place to read in binary data?

I think something like ``NonlinearSolver::Binary_input:`` is a reasonable choice.



> Is my understanding correct here?

yes.



> Is it possible to use boosts archive functionality to store only the time, time step, time step number, and etc, to a separate file at each time step?

Sure. I would create a small struct with the relevant data, fill it with the current information, and then serialize it.